march_hardware/include/march_hardware/imotioncube/imotioncube.h

```diff
@@ -120,7 +120,7 @@ class IMotionCube : public Slave
    * @param end_address the found end address of the setup in the .sw file
    * @return the computed checksum in the .sw file
    */
-  uint32_t computeSWCheckSum(int& start_address, int& end_address);
+  uint16_t computeSWCheckSum(int& start_address, int& end_address);
```

Comment:
Shouldn't the start and end address also be `uint16_t`? Since you read those from the file and also use to write an SDO.
```suggestion
  uint16_t computeSWCheckSum(uint16_t& start_address, uint16_t& end_address);
```